### PR TITLE
feature/allow-querystring-in-mqttwebsocketconfig

### LINKED ISSUE #335: Allow querystring in MqttWebsocketConfig
## Problem or use case
Hi,
when I try use websockets that need queryString I discovered that the request is bad formatted because the MqttWebSocketInitializer is using URI class only which the serverPath argument.

## Preferred solution or suggestions
Add queryString in MqttWebSocketConfig.
Use queryString of the new MqttWeSocketConfig in URI inside of initChannel of MqttWebSocketInitializer class.

Thanks

### PR DESCRIPTION
**Pull request by alsanrum**
**Motivation**
Allow querystring in MqttWebsocketConfig
Resolves #335 

**Changes**
Add queryString in MqttWebSocketConfig.
Use queryString of the new MqttWeSocketConfig in URI inside of initChannel of MqttWebSocketInitializer class.

### CODE REVIEW

**Comment by SgtSilvio** [comment]
@cla-bot check

**Comment by SgtSilvio** [comment]
Resolves #335

File: src/main/java/com/hivemq/client/internal/mqtt/MqttWebSocketConfigImplBuilder.java [hunk 1]
```diff
@@ -49,13 +51,18 @@
         return self();
     }
 
+    public @NotNull B queryString(final @Nullable String queryString) {
```
Comment (SgtSilvio) [inline src/main/java/com/hivemq/client/internal/mqtt/MqttWebSocketConfigImplBuilder.java]: This method is missing in the `MqttWebSocketConfigBuilderBase` interface

File: src/main/java/com/hivemq/client/internal/mqtt/MqttWebSocketConfigImpl.java [hunk 2]
```diff
@@ -27,21 +27,28 @@
...
 
     private final @NotNull String serverPath;
+    private final @NotNull String queryString;
     private final @NotNull String subprotocol;
 
-    MqttWebSocketConfigImpl(final @NotNull String serverPath, final @NotNull String subprotocol) {
+    MqttWebSocketConfigImpl(final @NotNull String serverPath, final @NotNull String queryString, final @NotNull String subprotocol) {
         this.serverPath = serverPath;
         this.subprotocol = subprotocol;
+        this.queryString = queryString;
```
Comment (SgtSilvio) [inline src/main/java/com/hivemq/client/internal/mqtt/MqttWebSocketConfigImpl.java]: should be the same ordering as the fields itself: `serverPath`, `queryString`, `subprotocol`

File: src/main/java/com/hivemq/client/internal/mqtt/MqttWebSocketConfigImplBuilder.java [hunk 3]
```diff
@@ -49,13 +51,18 @@
...
+        this.queryString = Checks.notNull(subprotocol, "QueryString");
```
Comment (SgtSilvio) [inline src/main/java/com/hivemq/client/internal/mqtt/MqttWebSocketConfigImplBuilder.java]:
```suggestion
        this.queryString = Checks.notNull(subprotocol, "Query string");
```

File: src/main/java/com/hivemq/client/internal/mqtt/MqttWebSocketConfigImplBuilder.java (hunk 1 above)
Comment (alsanrum) [reply src/main/java/com/hivemq/client/internal/mqtt/MqttWebSocketConfigImplBuilder.java]: Done

File: src/main/java/com/hivemq/client/internal/mqtt/MqttWebSocketConfigImpl.java (hunk 2 above)
Comment (alsanrum) [reply src/main/java/com/hivemq/client/internal/mqtt/MqttWebSocketConfigImpl.java]: Done

File: src/main/java/com/hivemq/client/internal/mqtt/MqttWebSocketConfigImplBuilder.java (hunk 3 above)
Comment (alsanrum) [reply src/main/java/com/hivemq/client/internal/mqtt/MqttWebSocketConfigImplBuilder.java]: Done

**Comment by SgtSilvio** [comment]
As this PR adds new APIs, it will be part of the next minor version 1.2.0.
So we can not merge this directly to develop yet, but instead to `develop-1-2`.
The feature can then already be used using a snapshot version `develop-1-2-SNAPSHOT` via JitPack: https://hivemq.github.io/hivemq-mqtt-client/docs/installation.html#snapshots

**Pull request merged**